### :x: 10 Tests Failed:
| Tests completed | Failed | Passed | Skipped |
|---|---|---|---|
| 10811 | 10 | 10801 | 52 |
<details><summary>View the full list of 10 :snowflake: flaky test(s)</summary>

> <pre><code class="language-python">github.com/steveyegge/gastown/internal/cmd::TestFilterAndSortSessions_SortOrder</code></pre>
> **Flake rate in main:** 88.89% (Passed 3 times, Failed 24 times)
> <details><summary>Stack Traces | 0s run time</summary>
> 
> > <pre><code class="language-python">=== RUN   TestFilterAndSortSessions_SortOrder
> >     agents_test.go:323: filterAndSortSessions returned 7 agents, want 8
> > --- FAIL: TestFilterAndSortSessions_SortOrder (0.00s)
> > </code></pre>
> 
> </details>


> <pre><code class="language-python">github.com/steveyegge/gastown/internal/cmd::TestGetTrackedIssues_FallsBackToShowTrackedDependencies</code></pre>
> **Flake rate in main:** 100.00% (Passed 0 times, Failed 21 times)
> <details><summary>Stack Traces | 0s run time</summary>
> 
> > <pre><code class="language-python">=== RUN   TestGetTrackedIssues_FallsBackToShowTrackedDependencies
> >     convoy_external_tracking_test.go:98: getTrackedIssues: querying tracked issues for hq-cv-ext: bd dep: unexpected bd args: dep list hq-cv-ext --direction=down --type=tracks --json
> > --- FAIL: TestGetTrackedIssues_FallsBackToShowTrackedDependencies (0.00s)
> > </code></pre>
> 
> </details>


> <pre><code class="language-python">github.com/steveyegge/gastown/internal/cmd::TestGuessSessionFromWorkerDir</code></pre>
> **Flake rate in main:** 88.89% (Passed 3 times, Failed 24 times)
> <details><summary>Stack Traces | 0s run time</summary>
> 
> > <pre><code class="language-python">=== RUN   TestGuessSessionFromWorkerDir
> > --- FAIL: TestGuessSessionFromWorkerDir (0.00s)
> > </code></pre>
> 
> </details>


> <pre><code class="language-python">github.com/steveyegge/gastown/internal/cmd::TestGuessSessionFromWorkerDir/different_rig</code></pre>
> **Flake rate in main:** 88.89% (Passed 3 times, Failed 24 times)
> <details><summary>Stack Traces | 0s run time</summary>
> 
> > <pre><code class="language-python">=== RUN   TestGuessSessionFromWorkerDir/different_rig
> >     agents_test.go:626: guessSessionFromWorkerDir(&quot;.../myrig/crew/alpha&quot;, &quot;/town&quot;) = &quot;gt-crew-alpha&quot;, want &quot;mr-crew-alpha&quot;
> > --- FAIL: TestGuessSessionFromWorkerDir/different_rig (0.00s)
> > </code></pre>
> 
> </details>


> <pre><code class="language-python">github.com/steveyegge/gastown/internal/cmd::TestJSONOutput_ErrorsReturnNonZeroExit</code></pre>
> **Flake rate in main:** 88.89% (Passed 3 times, Failed 24 times)
> <details><summary>Stack Traces | 0.01s run time</summary>
> 
> > <pre><code class="language-python">=== RUN   TestJSONOutput_ErrorsReturnNonZeroExit
> >     convoy_stage_test.go:2519: error output should still be valid JSON: unexpected end of JSON input
> >         raw:
> > --- FAIL: TestJSONOutput_ErrorsReturnNonZeroExit (0.01s)
> > </code></pre>
> 
> </details>


> <pre><code class="language-python">github.com/steveyegge/gastown/internal/cmd::TestJSONOutput_NoHumanReadableText</code></pre>
> **Flake rate in main:** 88.89% (Passed 3 times, Failed 24 times)
> <details><summary>Stack Traces | 0.03s run time</summary>
> 
> > <pre><code class="language-python">=== RUN   TestJSONOutput_NoHumanReadableText
> >     convoy_stage_test.go:2463: stdout is not valid JSON: unexpected end of JSON input
> >         raw:
> > --- FAIL: TestJSONOutput_NoHumanReadableText (0.03s)
> > </code></pre>
> 
> </details>


> <pre><code class="language-python">github.com/steveyegge/gastown/internal/cmd::TestRunConvoyList_UsesTownRootAndStripsBeadsDir</code></pre>
> **Flake rate in main:** 100.00% (Passed 0 times, Failed 21 times)
> <details><summary>Stack Traces | 0.01s run time</summary>
> 
> > <pre><code class="language-python">=== RUN   TestRunConvoyList_UsesTownRootAndStripsBeadsDir
> > ⚠ Warning: skipping convoy hq-cv-town: querying tracked issues for hq-cv-town: bd dep: unexpected bd args: dep list hq-cv-town --direction=down --type=tracks --json
> >     convoy_bd_routing_test.go:156: expected convoy JSON output, got:
> >         []
> > --- FAIL: TestRunConvoyList_UsesTownRootAndStripsBeadsDir (0.01s)
> > </code></pre>
> 
> </details>


> <pre><code class="language-python">github.com/steveyegge/gastown/internal/cmd::TestRunConvoyStatus_UsesTownRootAndStripsBeadsDir</code></pre>
> **Flake rate in main:** 100.00% (Passed 0 times, Failed 21 times)
> <details><summary>Stack Traces | 0.01s run time</summary>
> 
> > <pre><code class="language-python">=== RUN   TestRunConvoyStatus_UsesTownRootAndStripsBeadsDir
> >     convoy_bd_routing_test.go:211: runConvoyStatus: getting tracked issues for hq-cv-status: querying tracked issues for hq-cv-status: bd dep: unexpected bd args: dep list hq-cv-status --direction=down --type=tracks --json
> > --- FAIL: TestRunConvoyStatus_UsesTownRootAndStripsBeadsDir (0.01s)
> > </code></pre>
> 
> </details>


> <pre><code class="language-python">github.com/steveyegge/gastown/internal/polecat::TestManagerAgentLifecycleUsesTownBeadsDir</code></pre>
> **Flake rate in main:** 100.00% (Passed 0 times, Failed 21 times)
> <details><summary>Stack Traces | 0.02s run time</summary>
> 
> > <pre><code class="language-python">=== RUN   TestManagerAgentLifecycleUsesTownBeadsDir
> >     manager_test.go:2178: manager create did not use town BEADS_DIR; log:
> >         env=.../TestManagerAgentLifecycleUsesTownBeadsDir3414560806/001/.beads args=init --prefix gt --server
> >         env=.../TestManagerAgentLifecycleUsesTownBeadsDir3414560806/001/.beads args=config set issue_prefix gt
> >         env=.../TestManagerAgentLifecycleUsesTownBeadsDir3414560806/001/.beads args=migrate --yes
> >         env=.../TestManagerAgentLifecycleUsesTownBeadsDir3414560806/001/.beads args=config set types.custom agent,role,rig,convoy,slot,queue,event,message,molecule,gate,merge-request
> >         env=.../TestManagerAgentLifecycleUsesTownBeadsDir3414560806/001/.beads args=config get types.custom
> >         env=.../TestManagerAgentLifecycleUsesTownBeadsDir3414560806/001/.beads args=--allow-stale version
> >         env=.../TestManagerAgentLifecycleUsesTownBeadsDir3414560806/001/.beads args=create --json --id=gt-gastown-polecat-rust --title=gt-gastown-polecat-rust --description=gt-gastown-polecat-rust
> >         
> >         role_type: polecat
> >         rig: gastown
> >         agent_state: spawning
> >         hook_bead: null
> >         cleanup_status: null
> >         active_mr: null
> >         notification_level: null --type=task --labels=gt:agent --force
> >         env=.../TestManagerAgentLifecycleUsesTownBeadsDir3414560806/001/.beads args=show gt-gastown-polecat-rust --json
> >         env=.../TestManagerAgentLifecycleUsesTownBeadsDir3414560806/001/.beads args=update gt-gastown-polecat-rust --description=gt-gastown-polecat-rust
> >         
> >         role_type: polecat
> >         rig: gastown
> >         agent_state: nuked
> >         hook_bead: null
> >         cleanup_status: null
> >         active_mr: null
> >         notification_level: null
> > --- FAIL: TestManagerAgentLifecycleUsesTownBeadsDir (0.02s)
> > </code></pre>
> 
> </details>


> <pre><code class="language-python">github.com/steveyegge/gastown/internal/refinery::TestEngineerClearAgentActiveMRUsesTownBeadsDir</code></pre>
> **Flake rate in main:** 100.00% (Passed 0 times, Failed 21 times)
> <details><summary>Stack Traces | 0.01s run time</summary>
> 
> > <pre><code class="language-python">=== RUN   TestEngineerClearAgentActiveMRUsesTownBeadsDir
> >     engineer_test.go:154: refinery active_mr cleanup did not use town BEADS_DIR; log:
> >         env=.../TestEngineerClearAgentActiveMRUsesTownBeadsDir2600610266/001/.beads args=--allow-stale version
> >         env=.../TestEngineerClearAgentActiveMRUsesTownBeadsDir2600610266/001/.beads args=show gt-gastown-polecat-rust --json
> >         env=.../TestEngineerClearAgentActiveMRUsesTownBeadsDir2600610266/001/.beads args=update gt-gastown-polecat-rust --description=gt-gastown-polecat-rust
> >         
> >         role_type: polecat
> >         rig: gastown
> >         agent_state: idle
> >         hook_bead: null
> >         cleanup_status: null
> >         active_mr: null
> >         notification_level: null
> > --- FAIL: TestEngineerClearAgentActiveMRUsesTownBeadsDir (0.01s)
> > </code></pre>
> 
> </details>

</details>

To view more test analytics, go to the [Test Analytics Dashboard](https://app.codecov.io/gh/gastownhall/gastown/tests/rbriski%2Fgastown%3Afix%2Ftest-config-export-auto)
<sub>📋 Got 3 mins? [Take this short survey](https://forms.gle/22i53Qa1CySZjA6c7) to help us improve Test Analytics.</sub>

--- a/internal/cmd/convoy_bd_routing_test.go
+++ b/internal/cmd/convoy_bd_routing_test.go
@@ -112,7 +112,7 @@ case "$*" in
 	  "list --json --limit=0 --all --flat")
 	    echo '[]'
 	    ;;
-  "dep list hq-cv-town --direction=down --type=tracks --allow-stale --json")
+  "dep list hq-cv-town --direction=down --type=tracks --json")
     if [ "$PWD" != "%s" ]; then
       echo "expected town root, got $PWD" >&2
       exit 1
@@ -185,7 +185,7 @@ case "$*" in
     fi
     echo '[{"id":"hq-cv-status","title":"Status convoy","status":"open","issue_type":"convoy","created_at":"2026-03-09T00:00:00Z","labels":[],"dependencies":[]}]'
     ;;
-  "dep list hq-cv-status --direction=down --type=tracks --allow-stale --json")
+  "dep list hq-cv-status --direction=down --type=tracks --json")
     if [ "$PWD" != "%s" ]; then
       echo "expected town root, got $PWD" >&2
       exit 1

--- a/internal/cmd/convoy_external_tracking_test.go
+++ b/internal/cmd/convoy_external_tracking_test.go
@@ -70,7 +70,7 @@ case "$*" in
   "--allow-stale version")
     exit 0
     ;;
-  "dep list hq-cv-ext --direction=down --type=tracks --allow-stale --json")
+  "dep list hq-cv-ext --direction=down --type=tracks --json")
     echo '[]'
     ;;
   "show hq-cv-ext --json")

--- a/internal/cmd/convoy_launch_test.go
+++ b/internal/cmd/convoy_launch_test.go
@@ -742,7 +742,7 @@ case "$*" in
     # bdDepListRawIDs down: return tracked bead IDs
     echo '[{"depends_on_id":"external:ghostty:ghostty-1i4.3"},{"depends_on_id":"external:ghostty:ghostty-1i4.4"}]'
     ;;
-  "dep list hq-cv-ext --direction=down --type=tracks --allow-stale --json")
+  "dep list hq-cv-ext --direction=down --type=tracks --json")
     echo '[{"id":"external:ghostty:ghostty-1i4.3"},{"id":"external:ghostty:ghostty-1i4.4"}]'
     ;;
   "dep list hq-cv-ext --json")

--- a/internal/cmd/statusline_test.go
+++ b/internal/cmd/statusline_test.go
@@ -15,6 +15,7 @@ func setupCmdTestRegistry(t *testing.T) {
 	registry := session.NewPrefixRegistry()
 	registry.Register("gt", "gastown")
 	registry.Register("do", "coder_dotfiles")
+	registry.Register("mr", "myrig")
 	old := session.DefaultRegistry()
 	session.SetDefaultRegistry(registry)
 	t.Cleanup(func() { session.SetDefaultRegistry(old) })

--- a/internal/polecat/manager_test.go
+++ b/internal/polecat/manager_test.go
@@ -2174,10 +2174,10 @@ esac
 	if strings.Contains(logOutput, "env="+rigBeadsDir) {
 		t.Fatalf("manager agent lifecycle used rig BEADS_DIR; log:\n%s", logOutput)
 	}
-	if !strings.Contains(logOutput, "env="+townBeadsDir+" args=") || !strings.Contains(logOutput, " create") {
+	if !strings.Contains(logOutput, "env="+townBeadsDir+" args=") || !strings.Contains(logOutput, "args=create") {
 		t.Fatalf("manager create did not use town BEADS_DIR; log:\n%s", logOutput)
 	}
-	if !strings.Contains(logOutput, "env="+townBeadsDir+" args=") || !strings.Contains(logOutput, " show") || !strings.Contains(logOutput, " update") {
+	if !strings.Contains(logOutput, "env="+townBeadsDir+" args=") || !strings.Contains(logOutput, "args=show") || !strings.Contains(logOutput, "args=update") {
 		t.Fatalf("manager reset did not use town BEADS_DIR for show/update; log:\n%s", logOutput)
 	}
 }

--- a/internal/refinery/engineer_test.go
+++ b/internal/refinery/engineer_test.go
@@ -150,7 +150,7 @@ esac
 	if strings.Contains(logOutput, "env="+rigBeadsDir) {
 		t.Fatalf("refinery active_mr cleanup used rig BEADS_DIR; log:\n%s", logOutput)
 	}
-	if !strings.Contains(logOutput, "env="+townBeadsDir+" args=") || !strings.Contains(logOutput, " show") || !strings.Contains(logOutput, " update") {
+	if !strings.Contains(logOutput, "env="+townBeadsDir+" args=") || !strings.Contains(logOutput, "args=show") || !strings.Contains(logOutput, "args=update") {
 		t.Fatalf("refinery active_mr cleanup did not use town BEADS_DIR; log:\n%s", logOutput)
 	}
 }

